tests/framework/extensions/provisioning/creates.go

```diff
@@ -227,23 +227,30 @@ func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvide
 		}
 	}
 
-	if clustersConfig.Hardened && clustersConfig.KubernetesVersion <= string(provisioninginput.HardenedKubeVersion) {
-		err = hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
-		if err != nil {
-			return nil, err
-		}
+	if clustersConfig.Hardened {
+		var hardenCluster *apiv1.Cluster
+		if strings.Contains(clustersConfig.KubernetesVersion, clusters.K3SClusterType.String()) {
+			err = k3sHardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode, clustersConfig.KubernetesVersion)
+			if err != nil {
+				return nil, err
+			}
+
+			hardenCluster = clusters.HardenK3SClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+		} else {
+			err = rke2Hardening.HardeningNodes(client, clustersConfig.Hardened, nodes, rolesPerNode)
+			if err != nil {
+				return nil, err
+			}
 
-		hardenCluster := clusters.HardenK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
+			hardenCluster = clusters.HardenRKE2ClusterConfig(clusterName, namespace, clustersConfig, nil, "")
```

Comment:
is line 238 really any different from  line 245? Why not just move it out with the update call. And we can remove the var at the top

Reply:
Discussed offline, this is good